validator_client/src/block_service.rs

```diff
@@ -335,20 +335,30 @@ impl<T: SlotClock + 'static, E: EthSpec> BlockService<T, E> {
             )
         }
 
-        for validator_pubkey in proposers {
-            let builder_proposals = self
-                .validator_store
-                .get_builder_proposals(&validator_pubkey);
-            let service = self.clone();
-            let log = log.clone();
-            self.inner.context.executor.spawn(
-                async move {
-                    if builder_proposals {
+        let deneb_fork_activated = self
+            .context
+            .eth2_config
+            .spec
+            .deneb_fork_epoch
+            .and_then(|fork_epoch| {
+                let current_epoch = self.slot_clock.now()?.epoch(E::slots_per_epoch());
+                Some(current_epoch >= fork_epoch)
+            })
+            .unwrap_or(false);
+
+        if !self.validator_store.produce_block_v3() || deneb_fork_activated {
```

Comment:
should this be `self.validator_store.produce_block_v3()` instead? (without the `!`)